magento/magento2
# Remove media gallery assets metadata when a directory removed

### Description (*)
This PR introduces a logic for removing meta-information from media gallery (database records) once a directory with media assets has been removed.

### Fixed Issues (if relevant)
1. https://github.com/magento/adobe-stock-integration/issues/825 : The image state is not updated when the containing folder has been deleted

### Manual testing scenarios (*)
- Install Magento with Adobe Stock Integration
- Configure the integration in Stores -> Configuration -> System -> Adobe Stock Integration fieldset

1. From Magento Admin go to Content - Pages, click **Add New Page**
2. Expand **Content** Click **Show / Hide Editor**, select **Insert Image...**
3. Select Storage Root and click **Create Folder**, name the folder and click OK
4. Select the previously created folder and Click **Search Adobe Stock**
5. Select an image from Adobe Stock and **Save Preview**, Confirm
6. Select the previously created folder and click **Delete Folder**
7. Click **Search Adobe Stock** again and observe the previously saved image


The "Save Preview" and "License and Save" buttons should be present and work as expected
